app/streams/publishing_api/message_processor.rb

```diff
@@ -56,7 +56,7 @@ def handle_existing(item)
   def handle_new_version(item)
     new_item = item.copy_to_new_version!(base_path: base_path, payload_version: payload_version)
 
-    Items::Jobs::ImportContentDetailsJob.perform_async(new_item.id, current_date.day, current_date.month, current_date.year)
+    Items::Importers::ContentDetails.run(new_item.id, current_date.day, current_date.month, current_date.year)
```

Comment:
I'm happy with this change but just wondering if we should do the same with the quality metrics job.

This would mean the unit of work is always one content item, and if everything fails we retry that whole content item from scratch.

I think this makes the system simpler to reason about overall.

Reply:
What are your thoughts about dealing with errors on the PAAS service that we are invoking to retrieve the quality metrics. The system could be down, and we would need rollback all changes to the dataware dimensions...

Reply:
I'm not sure yet. I think this applies anywhere we do something that has the potential to leave the data in an incomplete state.

One way might be to have an explicit [staging area](https://en.wikipedia.org/wiki/Staging_(data)), then at the end of processing an item you can store the complete data in one transaction.

Reply:
As discussed offline, we will have a more in-depth discussion about it in the next developer's session.